File: app/src/main/java/pe/com/gadolfolozano/mymovie/ui/main/MainActivityModule.java
package pe.com.gadolfolozano.mymovie.ui.main;

import dagger.Module;
import dagger.Provides;
import pe.com.gadolfolozano.mymovie.data.DataManager;

@Module
public class MainActivityModule {
    @Provides
    MainViewModel provideSplashViewModel(DataManager dataManager) {
        return new MainViewModel(dataManager);
    }
}

File: app/src/main/java/pe/com/gadolfolozano/mymovie/ui/main/MainActivityModule.kt
package pe.com.gadolfolozano.mymovie.ui.main

import dagger.Module
import dagger.Provides
import pe.com.gadolfolozano.mymovie.data.DataManager

@Module
class MainActivityModule {
    @Provides
    internal fun provideSplashViewModel(dataManager: DataManager): MainViewModel {
        return MainViewModel(dataManager)
    }
}
